### ISSUE TITLE
Migrate target_collision() from deprecated affinity to metric parameter in AgglomerativeClustering

### ISSUE THREAD

**Issue by monodera** [Subaru-PFS/spt_target_uploader#421]
## Summary

The `target_collision()` function in `ppp.py` uses the deprecated `affinity` parameter with scikit-learn's `AgglomerativeClustering`. This needs to be migrated to use the `metric` parameter.

## Background

- Reported in PR #418 code review
- The `affinity` parameter was deprecated in scikit-learn 1.2 and removed in 1.4
- The project requires `scikit-learn>=1.3.0`
- There is already a FIXME comment documenting this issue

## Current Code

```python
# ppp.py:478-486
# FIXME: `affinity` was deprecated in sklearn 1.2 and has been removed in 1.4.
# Use `metric` instead, but also check if "haversine" is supported.
# See https://scikit-learn.org/stable/modules/generated/sklearn.cluster.AgglomerativeClustering.html
db = AgglomerativeClustering(
    distance_threshold=np.radians(sep),
    n_clusters=None,
    affinity="haversine",  # <-- DEPRECATED
    linkage="single",
).fit(np.radians([sample["dec"], sample["ra"]]).T)
```

## Risk Assessment

**Low Risk** - This code path is currently unreachable:
- `target_collision()` is only called from `PFS_FoV()` when `mode == "KDE_collision"`
- No current calls to `PFS_FoV()` pass this mode (all use default `mode=None`)

## Required Changes

1. Replace `affinity="haversine"` with `metric="haversine"` (if supported) or find an alternative metric
2. Verify the haversine metric is supported in the current scikit-learn API
3. If haversine is not supported, implement an alternative approach (e.g., precomputed distance matrix)
4. Consider whether the `KDE_collision` mode should be removed if it's no longer needed

## References

- [scikit-learn AgglomerativeClustering documentation](https://scikit-learn.org/stable/modules/generated/sklearn.cluster.AgglomerativeClustering.html)
- PR #418 code review comment

**Comment by monodera** [comment]
This is not urgent, so maybe target for S27A or S27B CfP.

**Comment by monodera** [comment]
Fully addressed by #452 (commit 30b0d64), released in v3.5.0 and merged to `main`.

Rather than migrating `affinity=` → `metric=`, the dead code was removed: `target_collision()` was only reachable via `PFS_FoV(..., mode="KDE_collision")`, and no call site ever passed that mode. @wanqqq31 confirmed on the PR that dropping `KDE_collision` is safe (already removed from integration code); collision avoidance is handled downstream by ETS netflow (`collision_distance`, `elbow_collisions`).

Verified no remaining references to `affinity`/`target_collision`/`KDE_collision` anywhere in the tree. The issue stayed open only because the `Closes` keyword fired on a merge into `dev-main` rather than `main`.
